sources/lib/PropertyInfo/Extractor/TypeExtractor.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/*
+ * This file is part of the PommProject/PommBundle package.
+ *
+ * (c) 2018 Grégoire HUBERT <hubert.greg@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PommProject\PommBundle\PropertyInfo\Extractor;
+
+use PommProject\Foundation\Pomm;
+use PommProject\Foundation\Session;
+use Symfony\Component\PropertyInfo\PropertyTypeExtractorInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+/**
+ * Extract data using pomm.
+ *
+ * @package PommBundle
+ * @copyright 2018 Grégoire HUBERT
+ * @author Nicolas Joseph
+ * @license X11 {@link http://opensource.org/licenses/mit-license.php}
+ */
+class TypeExtractor implements PropertyTypeExtractorInterface
+{
+    private $pomm;
+
+    public function __construct(Pomm $pomm)
+    {
+        $this->pomm = $pomm;
+    }
+
+    /**
+     * @see PropertyTypeExtractorInterface
+     */
+    public function getTypes($class, $property, array $context = array())
+    {
+        if (isset($context['session:name'])) {
+            $session = $this->pomm->getSession($context['session:name']);
+        } else {
+            $session = $this->pomm->getDefaultSession();
+        }
+
+        if (isset($context['model:name'])) {
+            $model_name = $context['model:name'];
+        } else {
+            $model_name = "${class}Model";
```

Comment:
same thing, I prefer the
```php
$model_name = "{$class}Model";
```
syntax as it allows the use of sub properties or function calls